File: Game Code.js
// Game Code Updates

// Update RESTOCK_SECS
const RESTOCK_SECS = 180;

// Rarity system added to SEEDS
const SEEDS = {
    CARROT: { rarity: 'common' },
    TOMATO: { rarity: 'common' },
    SUNFLOWER: { rarity: 'uncommon' },
    SHADOWPETAL: { rarity: 'mythic' },
    VOIDBLOOM: { rarity: 'mythic' }
};

// Initialize stock in make_game_state
function make_game_state() {
    return {
        stock: [SEEDS.CARROT, SEEDS.TOMATO],
        // Other game state properties
    };
}

// Update do_restock function
function do_restock() {
    const restockItems = [];
    const chance = Math.random();
    if (chance < 0.2) { // 20% chance for shadowpetal
        restockItems.push(SEEDS.SHADOWPETAL);
    }
    if (chance < 0.02) { // 2% chance for voidbloom
        restockItems.push(SEEDS.VOIDBLOOM);
    }
    // Continue with restocking other items as needed
}

// Remove timer bar from draw_seed_stall function
function draw_seed_stall() {
    // Original drawing code
    // Removed timer bar implementation
}

// Change draw_shop to display numeric MM:SS countdown
function draw_shop(countdown) {
    const minutes = String(Math.floor(countdown / 60)).padStart(2, '0');
    const seconds = String(countdown % 60).padStart(2, '0');
    // Display countdown as MM:SS
    console.log(`Time remaining: ${minutes}:${seconds}`);
}

File: Game Code.py
# Updated Game Code

# Define constants
RESTOCK_SECS = 180  # Changed from 300 to 180 seconds

# Seed definitions with rarity levels
SEEDS = {
    "carrot": {"rarity": "common"},
    "tomato": {"rarity": "common"},
    "sunflower": {"rarity": "uncommon"},
    "shadowpetal": {"rarity": "mythic"},
    "voidbloom": {"rarity": "mythic"}
}

# Modify make_game_state function

def make_game_state():
    stock = {"carrot": 10, "tomato": 10}  # Initialize with only carrot and tomato
    return stock

# Update do_restock function

def do_restock():
    for seed, stock in stock.items():
        if seed == "voidbloom":
            if random.random() < 0.02:  # 2% chance
                stock += 1
        elif seed == "shadowpetal":
            if random.random() < 0.20:  # 20% chance
                stock += 1

# Remove the restock bar/timer display from draw_seed_stall function

def draw_seed_stall():
    # Previous display codes removed
    pass

# Change draw_shop to display numeric countdown instead of progress bar

def draw_shop(restock_time):
    print(f"Restock in: {restock_time} seconds")  # Numeric countdown

# Additional game logic here.